### PR TITLE
Tests for flipping signs on signed type edge case

### LINKED ISSUE #5103: Add tests for flipping the sign on (un)signed types for edge case values
From https://github.com/ethereum/solidity/pull/5087#discussion_r220715840:

Add end to end tests for:
```
    int x = -2**255;
    assert(-x == x);
```

### PR DESCRIPTION
**Pull request by Mordax**
FIxes #5103 

### Checklist
- [x] Code compiles correctly
- [?] All tests are passing //I pray they do
- [?] New tests have been created which fail without the change (if possible)
- [x] Used meaningful commit messages

### Description
I may have missed a few things, it's my first time working on a programming language. 
Since Issue #5103 's references EndToEndSuite, I made the test in the SolidityEndToEnd.cpp. 

Hopefully the integrated tests won't be too mad. If I'm missing any fundamental things, misunderstood something or if I should add another case in the same test, let me know. Cheers!

### CODE REVIEW

File: test/libsolidity/SolidityEndToEndTest.cpp [hunk 1]
```diff
@@ -14010,6 +14010,21 @@ BOOST_AUTO_TEST_CASE(test_underscore_in_hex)
...
+		contract test {
+			function f() public returns (bool){
+				int x = -2**255;
+				assert(-x == x);
+				return true;
+			}
+		}
+	)";
+	compileAndRun(sourceCode);
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(u256(0)));
```
Comment (chriseth) [inline test/libsolidity/SolidityEndToEndTest.cpp]: The function returns `true`, so this should be `encodeArgs(true)`. Apart from that, perfect!

**Review by chriseth — CHANGES_REQUESTED** [review]

File: test/libsolidity/SolidityEndToEndTest.cpp (hunk 1 above)
Comment (Mordax) [reply test/libsolidity/SolidityEndToEndTest.cpp]: Thanks for the feedback! I did a rebase though, not sure if that's the proper procedure here or if I should've just not updated and pushed the fix anyways. 

Edit: Fixed the rebase issue!

**Comment by chriseth** [comment]
Great! Could you squash the two commits into a single one? Then this is ready to be merged.

**Comment by Mordax** [comment]
Squashed the commits

**Review by chriseth — APPROVED** [review]

**Pull request merged**